Samples/outlook-check-item-categories/README.md

```diff
@@ -0,0 +1,222 @@
+    <LaunchEvent Type="OnNewMessageCompose" FunctionName="onItemComposeHandler"/>
+    <LaunchEvent Type="OnNewAppointmentOrganizer" FunctionName="onItemComposeHandler"/>
+    <LaunchEvent Type="OnMessageSend" FunctionName="onItemSendHandler" SendMode="SoftBlock"/>
+    <LaunchEvent Type="OnAppointmentSend" FunctionName="onItemSendHandler" SendMode="Block"/>
+</LaunchEvents>
+```
+
+The `OnMessageSend` event uses the `SoftBlock` option to prevent a user from sending a message if some or all of the required categories are missing. In the event that the add-in becomes unavailable (for example, if the add-in encounters a loading error) and can't check the message for applied categories, the user will still be able to send the message.
+
+The `OnAppointmentSend` event uses the `Block` option to prevent a user from sending a meeting invite if some or all of the required categories are missing. If the add-in encounters a loading error and can't check the invite for applied categories, the user will not be able to send the invite until the add-in becomes available again.
+
+For additional information on `SendMode` options, see [Available SendMode options](https://docs.microsoft.com//javascript/api/manifest/launchevent?view=common-js-preview#available-sendmode-options-preview).
+
+### Configure the event handlers
+
+The event object is passed to its respective handler in **commands.js** for processing. The `onItemComposeHandler` processes `onNewMessageCompose` and `onNewAppointmentOrganizer` events, while `onItemSendHandler` processes `onMessageSend` and `onAppointmentSend` events.
+
+The `associate` method is called in `commands.js` to map the function IDs specified in the manifest to the appropriate event handler.
+
+```javascript
+Office.actions.associate("onItemComposeHandler", onItemComposeHandler);
+Office.actions.associate("onItemSendHandler", onItemSendHandler);
+```
+
+Once the handler processes the event, it calls the `event.completed` method. This method uses the `allowEvent` property to indicate whether the handled event can continue to execute or must terminate. When the event must terminate because it doesn't meet the add-in's conditions, the `errorMessage` property is used to display an alert to the user.
+
+```javascript
+let message = `You must assign the following categories before your ${Office.context.mailbox.item.itemType} can be sent: ${requiredCategories.join(", ")}`;
+console.log(message);
+sendEvent.completed({ allowEvent: false, errorMessage: message });
+return;
+```
+
+![Screenshot of the Outlook Smart Alerts error message when required categories are missing from the message or appointment being sent.](./assets/outlook-check-item-categories-smart-alerts.png)
+
+Event-based add-ins are expected to be short-running and lightweight. To prevent the possibility of a long-running operation that may lead to an add-in timeout, the color category creation process is handled by `onItemComposeHandler` instead of `onItemSendHandler`.
```

Comment:
```suggestion
Event-based add-ins should be short-running and lightweight. To prevent the possibility of a long-running operation that may lead to an add-in timeout, the color category creation process is handled by `onItemComposeHandler` instead of `onItemSendHandler`.
```
Passive tense, plus who/what expects the event-based add-in to be short?